File: Lintcode/155-Minimum-Depth-Of-Binary-Tree/MinimumDepthOfBinaryTree.java

public class MinimumDepthOfBinaryTree {
	/**
     * @param root: The root of binary tree.
     * @return: An integer.
     */
    public int minDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int min = Integer.MAX_VALUE;
        if (root.left == null && root.right == null) {
            return 1;
        }
        if (root.left != null) {
            min = Math.min(min, minDepth(root.left));
        }
        if (root.right != null) {
            min = Math.min(min, minDepth(root.right));
        }
        return min + 1;
    }
}

File: Leetcode/111-Minimum-Depth-of-Binary-Tree/MinimumDepthOfBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    if (!root) {
        return 0
    }
    return getMin(root);
};

var getMin = function(node) {
    if (!node) {
        return Infinity
    }
    if (!node.left && !node.right) {
        return 1
    }
    return Math.min(getMin(node.left), getMin(node.right)) +1
};


